sdk/tests/js/tests/gl-object-get-calls.js

```diff
@@ -757,6 +757,32 @@ if (contextVersion > 1) {
         function(target) {
             return gl.getIndexedParameter(target, 0);
     });
+
+    debug("");
+    debug("Test getQueryParameter");
+    var buffer = gl.createBuffer();
+    gl.bindBuffer(gl.TRANSFORM_FEEDBACK_BUFFER, buffer);
+    gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 64, gl.DYNAMIC_DRAW);
+    var query = gl.createQuery();
+    gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, buffer);
+    gl.beginQuery(gl.TRANSFORM_FEEDBACK_PRIMITIVES_WRITTEN, query);
+    gl.beginTransformFeedback(gl.TRIANGLES);
+    gl.drawArrays(gl.POINTS, 0, 3);
+    gl.endTransformFeedback();
+    gl.endQuery(gl.TRANSFORM_FEEDBACK_PRIMITIVES_WRITTEN);
+    shouldBe('gl.getQueryParameter(query, gl.QUERY_RESULT_AVAILABLE)', 'true');
```

Comment:
It has not returned to the main loop properly. It must return to the main loop (potentially multiple times, using e.g. requestAnimationFrame) after gl.endQuery() before requiring that gl.getQueryParameter(query, gl.QUERY_RESULT_AVAILABLE) return true. The WebGL 2.0 spec now requires this in https://www.khronos.org/registry/webgl/specs/latest/2.0/#GET_QUERY_PARAMETER .